v2/parser/parse.go

```diff
@@ -42,43 +38,32 @@ type importPathString string
 // Builder lets you add all the go files in all the packages that you care
 // about, then constructs the type source data.
 type Builder struct {
-	context *build.Context
-
 	// If true, include *_test.go
 	IncludeTestFiles bool
 
-	// Map of package names to more canonical information about the package.
-	// This might hold the same value for multiple names, e.g. if someone
-	// referenced ./pkg/name or in the case of vendoring, which canonicalizes
-	// differently that what humans would type.
-	//
-	// This must only be accessed via getLoadedBuildPackage and setLoadedBuildPackage
-	buildPackages map[importPathString]*build.Package
-
-	fset *token.FileSet
-	// map of package path to list of parsed files
-	parsed map[importPathString][]parsedFile
-	// map of package path to absolute path (to prevent overlap)
-	absPaths map[importPathString]string
+	// Map of package paths to definitions.  These keys should be canonical
+	// Go import paths (example.com/foo/bar) and not local paths (./foo/bar).
+	goPkgs map[importPathString]*packages.Package
 
-	// Set by typeCheckPackage(), used by importPackage() and friends.
-	typeCheckedPackages map[importPathString]*tc.Package
-
-	// Map of package path to whether the user requested it or it was from
-	// an import.
+	// Keep track of which packages were directly requested (as opposed to
+	// those which are transitively loaded).
 	userRequested map[importPathString]bool
```

Comment:
How about using sets here? It would simplify `UserRequestedPackages()` etc. Or is it desirable to avoid a dependency on `k8s.io/utils` in gengo?

Reply:
I'd avoid the dep - "a little copy is better than a little dep" :)

Reply:
Also could make vendoring into k/k more messy